tests/common/mock_environment.py

```diff
@@ -20,10 +20,19 @@
 import shutil
 import subprocess
 
+from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.utils import fileutil
 from tests.tools import patch, patch_builtin
 
 
+class MockCommand:
```

Comment:
I am adding negative tests, so need failures when mocking commands.

I created a new class for mock commands (it used to be tuple <command>,<stdout>) to include the return value and stderr